src/PackageCompiler.jl

```diff
@@ -537,6 +541,7 @@ function create_sysimage(packages::Union{Nothing, Symbol, Vector{String}, Vector
                          soname=nothing,
                          compat_level::String="major",
                          extra_precompiles::String = "",
+                         precompile_execution_args::Cmd=``
```

Comment:
I really think this should _default to_ `-O0`. I think that in the normal case, users will all want to set this. You're right that in some corner cases they may have big datasets and want to use O2, but i think that is the exception, not the norm.

If you are worried about that being a breaking change, we could do it in two different PRs? Otherwise, i'd put it in this one.